### PR TITLE
Integration of cluster-up 

### PR DESCRIPTION
**Pull request by davidvossel**
This integrates cluster-up into the kubevirtci repo.

* Provisioning scripts are now all in ```cluster-provision``` dir
* Cluster up/down scripts are now all in ```cluster-up``` dir
* Introduces Make targets for cluster-up and cluster-down
* Top level README is now related to consuming clusters rather than provisioning them.

### CODE REVIEW

File: cluster-up/cluster/k8s-1.13.3/provider.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -e
+
+image="k8s-1.13.3@sha256:9c2b78e11c25b3fd0b24b0ed684a112052dff03eee4ca4bdcc4f3168f9a14396"
+
+source ${KUBEVIRTCI_PATH}/cluster/ephemeral-provider-common.sh
+
+function up() {
+    ${_cli} run $(_add_common_params)
```
Comment (cynepco3hahue) [inline cluster-up/cluster/k8s-1.13.3/provider.sh]: at least for k8s provider it a lot of duplication under the `up` function, maybe we can somehow to reuse it

File: cluster-up/cluster/okd-4.1.0-nodes3/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# OKD 4.1.0-rc.0 in ephemeral containers
```
Comment (cynepco3hahue) [inline cluster-up/cluster/okd-4.1.0-nodes3/README.md]: 4.1.0-nodes3

File: cluster-up/cluster/okd-4.1.0/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# OKD 4.1.0-rc.0 in ephemeral containers
```
Comment (cynepco3hahue) [inline cluster-up/cluster/okd-4.1.0/README.md]: remove all rc.0

File: cluster-up/cluster/okd-4.1.0-nodes3/provider.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+image="okd-4.1.0@sha256:5ff43c79338d19e550a9ca8ef67f09d82a3425590970607888249cb3ccae6ea8"
+
+source ${KUBEVIRTCI_PATH}/cluster/ephemeral-provider-common.sh
+
+function _port() {
+    ${_cli} ports --prefix $provider_prefix --container-name cluster "$@"
+}
+
+function up() {
+    params="--random-ports --background --prefix $provider_prefix --master-cpu 5 --workers-cpu 5 --registry-volume $(_registry_volume) kubevirtci/${image}"
```
Comment (cynepco3hahue) [inline cluster-up/cluster/okd-4.1.0-nodes3/provider.sh]: I believe you need to specify the number of nodes here, but anyway okd provider still does not support it, so I prefer to drop this provider until we will have a working solution to configure additional nodes under the OKD, in general, it should not be too hard, it only need to scale worker machineset.

File: cluster-up/cluster/okd-4.1.0/provider.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+set -e
+
+image="okd-4.1.0-rc.0@sha256:7b5badcf46fceae521706161622ee447608d8ff84f831729ea98e061da49f4d7"
```
Comment (cynepco3hahue) [inline cluster-up/cluster/okd-4.1.0/provider.sh]: remove all rc.0

File: cluster-up/tags [hunk 6]
```diff
@@ -0,0 +1,34 @@
+!_TAG_FILE_FORMAT	2	/extended format; --format=1 will not append ;" to lines/
```
Comment (cynepco3hahue) [inline cluster-up/tags]: can you explain what exactly this file is doing?

File: cluster-up/cluster/okd-4.1.0-nodes3/README.md (hunk 2 above)
Comment (davidvossel) [reply cluster-up/cluster/okd-4.1.0-nodes3/README.md]: ah, i need to remove this provider. It was an experiment I was working with. It isn't ready for consumption yet though.

File: cluster-up/cluster/okd-4.1.0-nodes3/provider.sh (hunk 4 above)
Comment (davidvossel) [reply cluster-up/cluster/okd-4.1.0-nodes3/provider.sh]: yep, i'll drop it for now. We can have this discussion later.

File: cluster-up/tags (hunk 6 above)
Comment (davidvossel) [reply cluster-up/tags]: ha, this is my ctags file used by my editor (vim).  It wasn't supposed to be committed.

File: cluster-up/cluster/k8s-1.13.3/provider.sh (hunk 1 above)
Comment (davidvossel) [reply cluster-up/cluster/k8s-1.13.3/provider.sh]: yep, i agree. I'd like to follow up with this to reduce the scope of the changes in this pr though.

**Comment by davidvossel** [comment]
pr updated.

- removed references to rc.0
- removed experimental nodes3 provider
- removed tags file that shouldn't have been committed.

File: cluster-up/oc.sh [hunk 7]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
```
Comment (rmohr) [inline cluster-up/oc.sh]: Looks like just an alias to `kubectl.sh`? Works for me this way.

File: cluster-up/ssh.sh [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        port=$(echo "$ports" | grep 2201 | awk -F':' '{print $2}')
+    fi
+
+    if [ -z "$port" ]; then
+        echo "no ssh port found for $node"
+        exit 1
+    fi
+    ssh -lcore -p $port core@127.0.0.1 -i ${ssh_key}
+else
+    ${_cli} --prefix $provider_prefix ssh "$1"
```
Comment (rmohr) [inline cluster-up/ssh.sh]: We should unify the access between okd and the other clusters. I understand that we are not that far yet.

File: cluster-up/cluster/okd-4.1.0/README.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
+# OKD 4.1.0 in ephemeral containers
```
[the remark this answers is not in the document]
Comment (davidvossel) [reply cluster-up/cluster/okd-4.1.0/README.md]: ssh works for okd

File: cluster-up/hack/config-default.sh [hunk 10]
```diff
@@ -0,0 +1,4 @@
+
+docker_prefix=${DOCKER_PREFIX:-kubevirt}
```
[the remark this answers is not in the document]
Comment (davidvossel) [reply cluster-up/hack/config-default.sh]: the dynamic configs are now placed in a directory outside of cluster-up

File: cluster-up/ssh.sh (hunk 8 above)
Comment (davidvossel) [reply cluster-up/ssh.sh]: agreed, that would be nice.

**Pull request merged**